src/ResourceManager/Sql/Commands.Sql/Elastic Pools/Cmdlet/SetAzureSqlElasticPool.cs

```diff
@@ -129,20 +176,91 @@ protected override IEnumerable<AzureSqlElasticPoolModel> ApplyUserInputToModel(I
         {
             string location = ModelAdapter.GetServerLocation(ResourceGroupName, ServerName);
             List<AzureSqlElasticPoolModel> newEntity = new List<AzureSqlElasticPoolModel>();
-            newEntity.Add(new AzureSqlElasticPoolModel()
+            AzureSqlElasticPoolModel newModel = new AzureSqlElasticPoolModel()
             {
                 ResourceGroupName = ResourceGroupName,
                 ServerName = ServerName,
-                Tags = TagsConversionHelper.ReadOrFetchTags(this, model.FirstOrDefault().Tags),
-                Location = location,
                 ElasticPoolName = ElasticPoolName,
-                DatabaseDtuMax = MyInvocation.BoundParameters.ContainsKey("DatabaseDtuMax") ? (int?)DatabaseDtuMax : null,
-                DatabaseDtuMin = MyInvocation.BoundParameters.ContainsKey("DatabaseDtuMin") ? (int?)DatabaseDtuMin : null,
-                Dtu = MyInvocation.BoundParameters.ContainsKey("Dtu") ? (int?)Dtu : null,
-                Edition = MyInvocation.BoundParameters.ContainsKey("Edition") ? (DatabaseEdition?)Edition : null,
-                StorageMB = MyInvocation.BoundParameters.ContainsKey("StorageMB") ? (int?)StorageMB : null,
+                Tags = TagsConversionHelper.CreateTagDictionary(Tags, validate: true),
+                Location = location,
                 ZoneRedundant = MyInvocation.BoundParameters.ContainsKey("ZoneRedundant") ? (bool?)ZoneRedundant.ToBool() : null,
-            });
+                MaxSizeBytes = MyInvocation.BoundParameters.ContainsKey("StorageMB") ? (long?)(StorageMB * Megabytes) : null
+            };
+
+            var elasticPool = ModelAdapter.GetElasticPool(ResourceGroupName, ServerName, ElasticPoolName);
+            Management.Sql.Models.Sku poolCurrentSku = elasticPool.Sku;
+            Management.Sql.Models.ElasticPoolPerDatabaseSettings poolCurrentDbSetting = elasticPool.PerDatabaseSettings;
+
+            if (ParameterSetName == DtuPoolParameterSet)
+            {
+                string edition = MyInvocation.BoundParameters.ContainsKey("Edition") ? Edition : poolCurrentSku.Tier;
+
+                newModel.Sku = new Management.Sql.Models.Sku()
+                {
+                    Name = AzureSqlElasticPoolAdapter.getPoolSkuName(edition),
+                    Tier = edition,
+                    Capacity = MyInvocation.BoundParameters.ContainsKey("Dtu") ? (int?)Dtu : null
+                };
+
+                // If update from Vcore pool to Dtu pool, PerDatabaseSettings will be set to:
+                //   Default if not provided
+                //   Values specified if provided.
+                // If update from Dtu to Dtu pool, PerDatabaseSettings will be keep same.
+                if (AzureSqlElasticPoolAdapter.isVcorePool(edition) && !AzureSqlElasticPoolAdapter.isVcorePool(poolCurrentSku.Tier) ||
```

Comment:
I don't think powershell should be doing logic like this. Why is it needed? If not specified then you should set null and I believe authorizer has logic to handle all the different cases here